File: ws/chicaiza/u2/ws12Inheritence/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/Location.java
package ec.edu.espe.farmsystem.model;

/**
 *
 * @author Mathews Pastor, Poower Rangers of Programing, @ESPE
 */
public class Location {
    private int xCoordinate;
    private int yCoordinate;

    @Override
    public String toString() {
        return "Location{" + "xCordinate=" + xCoordinate + ", yCordinate=" + yCoordinate + '}';
    }
    
    public Location(int xCordinate, int yCordinate) {
        this.xCoordinate = xCordinate;
        this.yCoordinate = yCordinate;
    }
    
    /**
     * @return the xCoordinate
     */
    public int getxCoordinate() {
        return xCoordinate;
    }

    /**
     * @param xCoordinate the xCoordinate to set
     */
    public void setxCoordinate(int xCoordinate) {
        this.xCoordinate = xCoordinate;
    }

    /**
     * @return the yCoordinate
     */
    public int getyCoordinate() {
        return yCoordinate;
    }

    /**
     * @param yCoordinate the yCoordinate to set
     */
    public void setyCoordinate(int yCoordinate) {
        this.yCoordinate = yCoordinate;
    }
    
    
}

File: hw/loza/u2/hw19Interface/Python/model/Location.py
class Location:
    def __init__(self, xCoordinate, yCoordinate):
        if type(self) is Location:
            raise TypeError("Location is an abstract class and cannot be instantiated directly.")
        self.xCoordinate = xCoordinate
        self.yCoordinate = yCoordinate

    def __str__(self):
        return f"""{{
        "xCoordinate": {self.xCoordinate},
        "yCoordinate": {self.yCoordinate}
        }}"""

    def get_x_coordinate(self):
        return self.xCoordinate

    def set_x_coordinate(self, xCoordinate):
        self.xCoordinate = xCoordinate

    def get_y_coordinate(self):
        return self.yCoordinate

    def set_y_coordinate(self, yCoordinate):
        self.yCoordinate = yCoordinate
